data_pipeline.py

```diff
@@ -0,0 +1,264 @@
+
+        images = tf.io.serialize_tensor(images)
+        depths = tf.io.serialize_tensor(depths)
+        extrinsics = tf.io.serialize_tensor(extrinsics)
+        intrinsic_matrix = tf.io.serialize_tensor(
+            np.loadtxt(intrinsic_matrix_path))
+        axis_matrix = tf.io.serialize_tensor(get_axismatrix(axis_matrix_path))
+        box_objects = tf.io.serialize_tensor(np.loadtxt(box_objects_path))
+
+        example = serialize_example(
+            images,
+            depths,
+            extrinsics,
+            intrinsic_matrix,
+            axis_matrix,
+            box_objects)
+        writer.write(example)
+
+
+def read_decode_tfrecords(serialized_example):
+    """Reads and parses tfrecords
+    Args:
+        serialized_example: path to save the tfrecord
+    Return: dictionary of data pipeline
+    """
+    feature_description = {
+        'images': tf.io.FixedLenFeature((), tf.string),
+        'depths': tf.io.FixedLenFeature((), tf.string),
+        'extrinsic_matrix': tf.io.FixedLenFeature((), tf.string),
+        'intrinsic_matrix': tf.io.FixedLenFeature((), tf.string),
+        'axis_matrix': tf.io.FixedLenFeature((), tf.string),
+        'box_objects': tf.io.FixedLenFeature((), tf.string),
+    }
+    example = tf.io.parse_single_example(
+        serialized_example, feature_description)
+
+    example['images'] = tf.io.parse_tensor(
+        example['images'], out_type=tf.float32)
+    example['depths'] = tf.io.parse_tensor(
+        example['depths'], out_type=tf.float32)
+    example['extrinsic_matrix'] = tf.io.parse_tensor(
+        example['extrinsic_matrix'], out_type=tf.float64)
+    example['intrinsic_matrix'] = tf.io.parse_tensor(
+        example['intrinsic_matrix'], out_type=tf.float64)
+    example['axis_matrix'] = tf.io.parse_tensor(
+        example['axis_matrix'], out_type=tf.float32)
+    example['box_objects'] = tf.io.parse_tensor(
+        example['box_objects'], out_type=tf.float64)
+
+    # Linearly scales each image in images to have mean 0 and variance 1.
+    example['images'] = tf.map_fn(
+        lambda image: tf.image.per_image_standardization(image),
+        example['images'])
+
+    return example
+
+
+def get_images_with_boxes(scene_dir):
+    """Get selected random images that have most boxes
+    Args:
+        scene_dir: directory path for scene
+    Return: image paths, depth_paths and extrinsic paths
+    """
+
+    image_paths = os.path.join(scene_dir, 'color')
+    depth_paths = os.path.join(scene_dir, 'depth')
+    extrinsic_paths = os.path.join(scene_dir, 'pose')
+
+    scene = os.path.basename(scene_dir)
+    axis_matrix = get_axismatrix(os.path.join(scene_dir, scene + '.txt'))
+    intrinsic_matrix = np.loadtxt(
+        os.path.join(
+            scene_dir,
+            'intrinsic_color.txt'))
+    box_objects = np.loadtxt(os.path.join(scene_dir, 'box_coords.txt'))
+
+    # assuming all images have the same width and height, change if not true
```

Comment:
Start your comment with the capital letters: Assuming, and end it with .
Also applied to the other comments in this PR as well.